cdc/model/changefeed.go

```diff
@@ -453,3 +453,40 @@ func (info *ChangeFeedInfo) fixMemoryQuota() {
 func (info *ChangeFeedInfo) fixScheduler() {
 	info.Config.FixScheduler()
 }
+
+type Barrier struct {
+	GlobalBarrierTs   Ts
+	TableBarrier      []TableBarrier
+	MinTableBarrierTs Ts
+}
+
+type TableBarrier struct {
+	ID        TableID
+	BarrierTs Ts
+}
+
+func NewBarrier(ts Ts) *Barrier {
+	return &Barrier{
+		GlobalBarrierTs: ts,
+	}
+}
+
+type DownStreamType int
+
+const (
+	// DB is the type of sink for database.
+	DB DownStreamType = iota
+	// Other is the type of kafka or storage.
+	Other
```

Comment:
What about `MQ`?

Reply:
MQ and cloud storage both belong to `Other` type.

Reply:
Prefer explicit, what about adding a new `Storage` type?

Reply:
done